week1/homework/src/server.js

```diff
@@ -10,10 +10,35 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
-  });
 
+    switch (request.url) {
+      case '/':
+        sendResponse(200, `this is ${state}`, response);
+        break;
+      case '/add':
+        sendResponse(200, `the state now is ${++state}`, response);
+        break;
+      case '/subtract':
+        sendResponse(200, `the state now is ${--state}`, response);
+        break;
+      case '/reset':
+        sendResponse(200, `the state return to ${(state = 10)}`, response);
+        break;
+      default:
+        sendResponse(404, `not found`, response);
+    }
+  });
   return server;
 }
+function sendResponse(statusNum, content, response) {
+  response.statusCode = statusNum;
+  response.setHeader('content-type', 'application/json');
```

Comment:
The response you're sending is not completely of the `content-type` `application/json`, you should not mix content types: only use JSON or plain text!